Add Query.node

marshalWilliam opened the pull request.

marshalWilliam committed:
Add Query.node

ralphcasipe1 commented on src/schemas/resolvers/resolver_query.ts:
```diff
@@ -1,29 +1,77 @@
+        result = await getProduct(id);
+        if (result) {
+          return {
+            __typename: "Product",
```
You can resolve the `__typename` via `Node.__resolveType` for example:
```ts
Node: {
  __resolveType(parent: <your type here>)  {
      if (id.includes('product', 32)) return 'Product';
      else if ((id.includes('account', 32)) return 'Account';
      else return;
   }
}
```

There's no need to throw an error for this, we just tell the end-user that there's no data found.

So that no need to wrap your returned queries in `result` variable

marshalWilliam committed:
Update#1 Query.node

Used  __resolveType instead of __typename in resolving Query.node

marshalWilliam committed:
Update#2 Query.node

Removed resolver fields for type Account and type Product except for Product.owner field
